Improve primitive intro (#2018)
Qiskit/documentation: docs/guides/primitives.mdx

one can create something useful for a given abstraction level.

The recent progress in quantum computing has increased the need to work at higher levels of abstraction.

> As we move toward larger quantum processing units (QPUs) and more complex workflows, the focus shifts from interacting with individual

Reviewer: ```suggestion
As the field moves toward larger quantum processing units (QPUs) and more complex workflows, the focus shifts from interacting with individual
```

qubit signals to viewing quantum devices as systems that perform tasks we need.

> The two most common tasks quantum computers are used for are sampling quantum states and calculating expectation values.

Reviewer: ```suggestion
The two most common tasks for quantum computers are sampling quantum states and calculating expectation values.
```

These tasks motivated the design of the Qiskit primitives: **Estimator** and **Sampler**.

- [Estimator](#estimator) computes expectation values of observables with respect to states prepared by quantum circuits.
- [Sampler](#sampler) samples the output register from quantum circuit execution.

In short, the computational model introduced by the Qiskit primitives moves quantum programming one step closer
to where classical programming is today, where the focus is less on the hardware details and more on the results
you are trying to achieve.

<Admonition type="note">

> The Qiskit SDK holds the definitions of the `Estimator` and `Sampler` primitives and acts  as the source-of-truth for this concept. It also contains reference implentations [(explained below)](./primitives#implementations) to assist providers of quantum compute resources in developing services which support Qiskit users.

Reviewer: ```suggestion
The Qiskit SDK holds the definitions of the `Estimator` and `Sampler` primitives and acts  as the source-of-truth for this concept. It also contains reference implementations [(explained below)](./primitives#implementations) to assist providers of quantum compute resources in developing services that support Qiskit users.
```

> For users of IBM Quantum hardware, the [`qiskit-ibm-runtime`](./index#qiskit-runtime) package possesses an implementation of these primitives which allow you to execute quantum workloads on IBM Quantum hardware.

Reviewer: ```suggestion
For users of IBM Quantum hardware, the [`qiskit-ibm-runtime`](./index#qiskit-runtime) package possesses an implementation of these primitives which allow you to execute quantum workloads on IBM Quantum&trade; hardware.
```

Reviewer: Note: we can now remove the IBM Quantum trademark farther down the page in the Next steps

</Admonition>

<span id="estimator"></span>

> ## Estimator

Reviewer 2: I think we should make it clearer that here we are talking about the runtime implementations of the primitives i know we mention it very briefly in the section above but its easy to miss. I think either update the heading or mention in the first paragraph of this section that this is demonstrating the runtime implementation. same deal for the sampler

one can create something useful for a given abstraction level.

The recent progress in quantum computing has increased the need to work at higher levels of abstraction.
As the field moves toward larger quantum processing units (QPUs) and more complex workflows, the focus shifts from interacting with individual
qubit signals to viewing quantum devices as systems that perform necessary tasks.

The two most common tasks for quantum computers are sampling quantum states and calculating expectation values.
These tasks motivated the design of the Qiskit primitives: **Estimator** and **Sampler**.

- Estimator computes expectation values of observables with respect to states prepared by quantum circuits.
- Sampler samples the output register from quantum circuit execution.

In short, the computational model introduced by the Qiskit primitives moves quantum programming one step closer
to where classical programming is today, where the focus is less on the hardware details and more on the results
you are trying to achieve.

## Primitive definition and implementations

> There are two types of Qiskit primitives: the base classes, and their implementations. The Qiskit primitives are defined by open-source primitive base classes that live in the Qiskit SDK (in the  [`qiskit.primitives`](/api/qiskit/primitives) module). Providers can use these base classes to derive their own Sampler and Estimator implementations.

Reviewer 2: I think it would be worth adding one more sentence to this paragraph that says that most users will interact with provider implementations, and only provider developers need to worry about the base primitives

> There are two types of Qiskit primitives: the base classes, and their implementations. The Qiskit primitives are defined by open-source primitive base classes that live in the Qiskit SDK (in the  [`qiskit.primitives`](/api/qiskit/primitives) module). Providers can use these base classes to derive their own Sampler and Estimator implementations.

Reviewer 2: ```suggestion
There are two types of Qiskit primitives: the base classes, and their implementations. The Qiskit primitives are defined by open-source primitive base classes that live in the Qiskit SDK (in the  [`qiskit.primitives`](/api/qiskit/primitives) module). Providers (such as Qiskit Runtime) can use these base classes to derive their own Sampler and Estimator implementations.
```

### Base classes

[`BaseEstimatorV2`](/api/qiskit/qiskit.primitives.BaseEstimatorV2) and [`BaseSamplerV2`](/api/qiskit/qiskit.primitives.BaseSamplerV2) - Abstract base classes that define a common interface for implementing primitives. All other classes in the [`qiskit.primitives`](/api/qiskit/primitives) module inherit from these base classes.  Developers should use these if they are interested in creating their own primitives-based execution model for a specific provider. These classes might also be useful for those who want to do highly customized processing and find that the existing primitives implementations are too simple for their needs.

<span id="implementations"></span>
### Implementations

> These are implementations of the the primitives base classes:

Reviewer 2: ```suggestion
These are implementations of the primitives base classes:
```

])
```

## Benefits of Qiskit primitives

> For Qiskit users, primitives let you write quantum code for a specific QPU without having to explicitly
> manage every detail. Also, because of the additional layer of abstraction, you might be able to more easily
> access advanced hardware capabilities of a given provider. For example, with Qiskit Runtime primitives,
> you can leverage the latest advancements in error mitigation and suppression by toggling options such as the primitive's [`resilience_level`,](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.EstimatorOptions#resilience_level) rather than building your own implementation of these techniques.

Reviewer: ```suggestion
With primitives, Qiskit users can write quantum code for a specific QPU without having to explicitly
manage every detail. Also, because of the additional layer of abstraction, you might be able to more easily
access advanced hardware capabilities of a given provider. For example, with Qiskit Runtime primitives,
you can take advantage of the latest advancements in error mitigation and suppression by toggling options such as the primitive's [`resilience_level`,](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.EstimatorOptions#resilience_level) rather than building your own implementation of these techniques.
```

For hardware providers, implementing primitives natively means you can provide your users with a more “out-of-the-box”

> way to access your hardware features (such as advanced post-processing technqiues). It is therefore easier for your users to benefit from your hardware's

Reviewer: ```suggestion
way to access your hardware features (such as advanced post-processing techniques). It is therefore easier for your users to benefit from your hardware's
```

best capabilities.

## Definition and implementation of Qiskit primitives

> The Qiskit primitives are defined by open-source primitive base-classes, from

Reviewer: ```suggestion
The Qiskit primitives are defined by open-source primitive base classes, from
```

which different providers can derive their own Sampler and Estimator implementations. Among the implementations
that use Qiskit, you can find reference primitive implementations for local simulation in the [`qiskit.primitives`](/api/qiskit/primitives) module.
Providers like [Qiskit Runtime](/api/qiskit-ibm-runtime/runtime_service) enable access to appropriate QPUs through native implementations of
their own primitives.

The [`qiskit.primitives`](/api/qiskit/primitives) module enables the development of primitive-style quantum programs and was specifically

> designed to simplify switching between different types of quantum compute resources. As described previously, there are two types of Qiskit primitives; the base class, and their implementations.

Reviewer: ```suggestion
designed to simplify switching between different types of quantum compute resources. As described previously, there are two types of Qiskit primitives: the base class, and their implementations.
```

### Base class

[`BaseEstimatorV2`](/api/qiskit/qiskit.primitives.BaseEstimatorV2) and [`BaseSamplerV2`](/api/qiskit/qiskit.primitives.BaseSamplerV2)

> These are abstract base classes that define a common interface for implementing primitives. All other classes in the [`qiskit.primitives`](/api/qiskit/primitives) module inherit from these base classes, and developers should use these if they are interested in developing their own primitives-based execution model for a specific provider. These classes might also be useful for those who want to do highly customized processing and find the existing primitives implementations too simple for their needs.

Reviewer: ```suggestion
These are abstract base classes that define a common interface for implementing primitives. All other classes in the [`qiskit.primitives`](/api/qiskit/primitives) module inherit from these base classes, and developers should use these if they are interested in developing their own primitives-based execution model for a specific provider. These classes might also be useful for those who want to do highly customized processing and find that the existing primitives implementations are too simple for their needs.
```

### Implementations

- [`StatevectorEstimator`](/api/qiskit/qiskit.primitives.StatevectorEstimator) and [`StatevectorSampler`](/api/qiskit/qiskit.primitives.StatevectorSampler#statevectorsampler)

> These classes are reference implementations of the primitives and use the simulator built into Qiskit. They leverage the Qiskit [`quantum_info`](/api/qiskit/quantum_info#quantum-information) module in the background, producing results based on ideal statevector simulations.

Reviewer: ```suggestion
These classes are reference implementations of the primitives and use the simulator built into Qiskit. They are built with the Qiskit [`quantum_info`](/api/qiskit/quantum_info#quantum-information) module in the background, producing results based on ideal statevector simulations.
```
Rewording because IBM Style says not to use leverage in tech language (and to limit it in marketing content)

])
```

> ## Benefits of Qiskit primitives

Reviewer 2: this section and the one after it are still confusing to me, it's not clear if we're talking about the runtime primitives specifically or the general concept of primitives

access advanced hardware capabilities of a given provider. For example, with Qiskit Runtime primitives,
you can take advantage of the latest advancements in error mitigation and suppression by toggling options such as the primitive's [`resilience_level`,](/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.EstimatorOptions#resilience_level) rather than building your own implementation of these techniques.

For hardware providers, implementing primitives natively means you can provide your users with a more “out-of-the-box”
way to access your hardware features (such as advanced post-processing techniques). It is therefore easier for your users to benefit from your hardware's
best capabilities.

## Definition and implementation of Qiskit primitives

The Qiskit primitives are defined by open-source primitive base classes, from
which different providers can derive their own Sampler and Estimator implementations. Among the implementations
that use Qiskit, you can find reference primitive implementations for local simulation in the [`qiskit.primitives`](/api/qiskit/primitives) module.

> Providers like [Qiskit Runtime](/api/qiskit-ibm-runtime/runtime_service) enable access to appropriate QPUs through native implementations of

Reviewer 2: I think this line is one of the most important and should be mentioned way earlier on. in fact this whole section is really important for understanding what primitives are and should come earlier before we launch into the specific descriptions of the runtime primitives

their own primitives.

The [`qiskit.primitives`](/api/qiskit/primitives) module enables the development of primitive-style quantum programs and was specifically

> designed to simplify switching between different types of quantum compute resources. As described previously, there are two types of Qiskit primitives: the base class, and their implementations.

Reviewer 2: it looks like theres some funny indenting here? 
<img width="872" alt="Screenshot 2024-10-08 at 1 03 53 PM" src="https://github.com/user-attachments/assets/11f37b2b-eddc-4bc1-9f70-3210e62d293e">
